src/utils/useTimeZoneUtils.ts

```diff
@@ -25,32 +35,40 @@ export const getUTC = (timeZones: TimeZoneDescription[]): TimeZoneDescription =>
 /**
  * Get a fallback time zone (based on org, user, etc.)
  */
-export const useDefaultTimeZone = (forEdit?: boolean) => {
+export const useDefaultTimeZone = () => {
   const { user } = useUser();
   const { selectedOrganization } = useOrganization();
   const timeZones = useTimeZones();
 
-  const orgTimeZone = getTimeZone(timeZones, selectedOrganization.timeZone);
-  if (orgTimeZone) {
-    return orgTimeZone;
-  }
+  return {
+    get: (forEdit?: boolean) => {
+      const orgTimeZone = getTimeZone(timeZones, selectedOrganization.timeZone);
+      if (orgTimeZone) {
+        return orgTimeZone;
+      }
 
-  const userTimeZone = forEdit ? getTimeZone(timeZones, user?.timeZone) : undefined;
-  if (userTimeZone) {
-    return userTimeZone;
-  }
+      const userTimeZone = forEdit ? getTimeZone(timeZones, user?.timeZone) : undefined;
+      if (userTimeZone) {
+        return userTimeZone;
+      }
 
-  return getUTC(timeZones);
+      return getUTC(timeZones);
+    },
+  };
 };
 
 /**
  * Get a fallback time zone for a location (based on location org, user, etc.)
  */
-export const useLocationTimeZone = (location: Location, forEdit?: boolean) => {
+export const useLocationTimeZone = () => {
   const timeZones = useTimeZones();
-  const defaultTimeZone = useDefaultTimeZone(forEdit);
+  const defaultTimeZone = useDefaultTimeZone();
 
-  return getTimeZone(timeZones, location.timeZone) ?? defaultTimeZone;
+  return {
+    get: (location?: Location, forEdit?: boolean) => {
```

Comment:
updated all the hooks with a currying function sort of - to enable delayed checking/fetching of time zones otherwise we can't use them with react states. now everything works clean.